employees/views.py

```diff
@@ -67,108 +63,60 @@ def perform_create(self, serializer: ReportSerializer) -> None:
+            form = ProjectJoinForm(
+                queryset=Project.objects.exclude(members__id=self.request.user.id).order_by("name"), data=request.POST
+            )
+            if form.is_valid():
+                project = Project.objects.get(id=int(self.request.POST["projects"]))
+                project.members.add(self.request.user)
+                project.full_clean()
+                project.save()
+                return self.form_valid_for_join_project()
+            else:
+                return self.form_invalid_for_join_project(form)
         else:
-            reports_serializer = ReportSerializer(context={"request": self.request})
-            reports_serializer.fields["date"].initial = str(datetime.datetime.now().date())
-        reports_serializer.fields["project"].queryset = Project.objects.filter(
-            members__id=self.request.user.id
-        ).order_by("name")
-        reports_serializer.fields["task_activities"].queryset = TaskActivityType.objects.order_by("name")
-        return reports_serializer
-
-    def _create_project_join_form(self) -> ProjectJoinForm:
-        project_form_queryset = Project.objects.exclude(members__id=self.request.user.id).order_by("name")
-        self.hide_join = not project_form_queryset.exists()
-        return ProjectJoinForm(queryset=project_form_queryset)
-
-    def initial(self, request: HttpRequest, *args: Any, **kwargs: Any) -> None:
-        logger.info(f"Initial method for user with id: {self.request.user.pk} in ReportList view")
-        super().initial(request, *args, **kwargs)
-        self.reports = self.get_queryset()
-        self.daily_hours_sum = self.reports.order_by().get_work_hours_sum_for_all_dates()
-        self.monthly_hours_sum = self.reports.order_by().get_work_hours_sum_for_all_authors()
-
-    def get(self, _request: HttpRequest) -> Response:
-        logger.info(f"User with id: {self.request.user.pk} get to the ReportList view")
-        return Response(
-            {
-                "serializer": self._create_serializer(),
-                "object_list": self.reports,
-                "daily_hours_sum": self.daily_hours_sum,
-                "monthly_hours_sum": self.monthly_hours_sum,
-                "UI_text": ReportListStrings,
-                "project_form": self._create_project_join_form(),
-                "hide_join": self.hide_join,
-            }
+            return super().post(request, *args, **kwargs)
+
+    def get_form(self, form_class: ReportForm = None) -> ReportForm:
+        if form_class is None:
+            form_class = super().get_form_class()
+        return form_class(user=self.request.user, **super().get_form_kwargs())
+
+    def get_context_data(self, **kwargs: Any) -> dict:
+        context_data = super().get_context_data(**kwargs)
+        context_data["object_list"] = (
+            super().get_queryset().filter(author=self.request.user).order_by("-date", "project__name", "-creation_date")
+        )
```

Comment:
Nice idea and it could work, so please try it. However I still think that ultimate solution to this is splitting the view into two - one for list, one for create.